### PR TITLE
Ruff mypy fixes

### PR DESCRIPTION
**Pull request by CatEek**
1. Proper checkpoint saving with state_dict for model, optim etc
2 Update mean/std from checkpoint
3. Create model refactor
4. Some ruff/mypy fixes
5. Tests update

### CODE REVIEW

**Comment by jdeschamps** [comment]
Is this branch superseeding https://github.com/CAREamics/careamics-restoration/pull/46 ?

**Review by jdeschamps — APPROVED** [review]

**Comment by jdeschamps** [comment]
The tests are not passing

**Review by jdeschamps — CHANGES_REQUESTED** [review]
In engine.py:65:

```python
config, model, state_dict=checkpoint
```

The smoke test gives the following error:
```
E       UnboundLocalError: local variable 'model' referenced before assignment
```

**Review by jdeschamps — CHANGES_REQUESTED** [review]
Due to changes to what the configuration model represents (path vs file name), the fixture in `conftest.py` are causing errors in the tests.

Check the `complete_config`

**Review by jdeschamps — CHANGES_REQUESTED** [review]
The tests relative to the model path in the Configuration are not passing. Most seem to be comparing a `Path` to a `str`.

File: src/careamics_restoration/dataset/tiling.py [hunk 1]
```diff
@@ -332,7 +334,7 @@ def extract_patches_predict(
     arr: np.ndarray,
     patch_size: Tuple[int],
     overlaps: Tuple[int],
-) -> Iterable[List[np.ndarray]]:
+) -> Iterable[Tuple[np.ndarray[Any, np.dtype[Any]], bool, Tuple[int, ...], Any, Any]]:
```
[the remark this answers is not in the document]
Comment (CatEek) [reply src/careamics_restoration/dataset/tiling.py:337]: Will take a look tomorrow

**Review by jdeschamps — APPROVED** [review]

**Pull request merged**